### PR TITLE
Map views => reshapes in aotautograd

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #84801

### CODE REVIEW

File: functorch/test/test_aotdispatcher.py [hunk 1]
```diff
@@ -0,0 +1,716 @@
+# Owner(s): ["module: functorch"]
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from torch.testing._internal.common_utils import TestCase, run_tests
```
Comment (ngimel) [inline functorch/test/test_aotdispatcher.py:9]: is this a new or renamed file?

**Review by ngimel — APPROVED** [review]

File: functorch/test/test_aotdispatcher.py (hunk 1 above)
Comment (Chillee) [reply functorch/test/test_aotdispatcher.py:9]: Yeah this is just `test_pythonkey.py` renamed.

**Comment by ezyang** [comment]
I'm not sure about this, mostly because reshape is getting decomposed so this transform is not idempotent; you will get increasingly long chains of .contiguous().contiguous() this way

**Comment by ngimel** [comment]
But `.contiguous` is idempotent, so we can have a pass trimming long `contiguous` chains? We can fix the current layer_norm + linear crisis by adding `.contiguous` into layer norm decomposition, and think about the proper solution, but generally, given that decompositions don't guarantee same striding behavior as eager (especially when eager is wrong, as it is with layer norm) we can't guarantee that views will always work after decompositions.

**Comment by Chillee** [comment]
How are we gonna have long `.contiguous()` chains? Contiguous isn't a `compositeexplicitautograd` op 🤔

**Comment by zou3519** [comment]
Why do we need to do this? Is it because AOTAutograd assumes contiguity of grad_out, so the trace isn't valid for inputs of other size/strides?

**Comment by ezyang** [comment]
I guess I'm generally looking for a unified story for how we will do cyclic decomps . Once we have it I'll probably be ok with this.

**Comment by ngimel** [comment]
> Why do we need to do this? Is it because AOTAutograd assumes contiguity of grad_out, so the trace isn't valid for inputs of other size/strides?

This is because decompositions are not guaranteed to produce same permutation as eager ops, so during retracing the original trace (based on eager) may be invalid. In the particular case that triggered this PR we had a layer norm op followed by nn.functional.linear. `nn.functional.linear` has different paths based on the contiguity of inputs. Eager layer norm produces contiguous output, so contiguous linear path was taken for original tracing (that calls `.view`), but decomposed layer norm can produce discontiguous output, and thus `view` operation in retracing fails.

**Comment by ngimel** [comment]
Yes we are assuming functionalization has happened.

**Comment by zou3519** [comment]
> Yes we are assuming functionalization has happened.

Cool, thanks for the context. No more concerns from me.

**Comment by desertfire** [comment]
Is there a conclusion on how to proceed with this PR? It does affects 12 TIMM models, https://github.com/pytorch/torchdynamo/issues/1135.

**Comment by ngimel** [comment]
Did we move pytorch pin? The dynamo issue should have been fixed by https://github.com/pytorch/pytorch/pull/84799

**Pull request closed without merging**